components/Users/Memberships/CancelMembership.js

```diff
@@ -91,39 +102,55 @@ export default class CancelPledge extends Component {
         cancellationType
       } = this.state
 
-      return mutation({
-        variables: {
-          membershipId: this.props.membership.id,
-          immediately,
-          suppressConfirmation,
-          suppressWinback,
-          details: {
-            reason,
-            type: cancellationType
+      if (this.props.cancellation) {
+        return mutation({
+          variables: {
+            cancellationId: this.props.cancellation.id,
```

Comment:
You could get rid of the if-else by doing something like:
```
return mutation({
  variables: {
    ...(this.props.cancellation && {
        cancellationId: this.props.cancellation.id
    }),
    ...(!this.props.cancellation && {
      membershipId: this.props.membership.id,
      immediately
    }),
    details: {
```

(or at least re-use the details object)